libr/include/r_util/r_pkcs7.h

```diff
@@ -0,0 +1,102 @@
+#ifndef R_PKCS7_H
+#define R_PKCS7_H
+
+typedef struct r_pkcs7_certificaterevocationlists_t {
+	ut32 length;
+	RX509CertificateRevocationList **elements;
+} RPKCS7CertificateRevocationLists;
+
+typedef struct r_pkcs7_extendedcertificatesandcertificates_t {
+	ut32 length;
+	RX509Certificate **elements;
+} RPKCS7ExtendedCertificatesAndCertificates;
+
+typedef struct r_pkcs7_digestalgorithmidentifiers_t {
+	ut32 length;
+	RX509AlgorithmIdentifier **elements;
+} RPKCS7DigestAlgorithmIdentifiers;
+
+typedef struct r_pkcs7_contentinfo_t {
+	RASN1String *contentType; //OID
+	RASN1Object *content; // optional. oid structure definition
+} RPKCS7ContentInfo;
+
+typedef struct r_pkcs7_issuerandserialnumber_t {
+	RX509Name issuer;
+	RASN1Object *serialNumber;
+} RPKCS7IssuerAndSerialNumber;
+
+typedef struct r_pkcs7_attribute_t {
+	RASN1String *oid; //OID
+	RASN1Object *data; // optional. oid structure definition
+} RPKCS7Attribute;
+
+typedef struct r_pkcs7_attributes_t {
+	ut32 length;
+	RPKCS7Attribute **elements;
+} RPKCS7Attributes;
+
+typedef struct r_pkcs7_signerinfo_t {
+	ut32 version;
+	RPKCS7IssuerAndSerialNumber issuerAndSerialNumber;
+	RX509AlgorithmIdentifier digestAlgorithm;
+	RPKCS7Attributes authenticatedAttributes; //Optional
+	RX509AlgorithmIdentifier digestEncryptionAlgorithm;
+	RASN1Object *encryptedDigest;
+	RPKCS7Attributes unauthenticatedAttributes; //Optional
+} RPKCS7SignerInfo;
+
+typedef struct r_pkcs7_signerinfos_t {
+	ut32 length;
+	RPKCS7SignerInfo **elements;
+} RPKCS7SignerInfos;
+
+typedef struct r_pkcs7_signeddata_t {
+	ut32 version;
+	RPKCS7DigestAlgorithmIdentifiers digestAlgorithms;
+	RPKCS7ContentInfo contentInfo;
+	RPKCS7ExtendedCertificatesAndCertificates certificates; //Optional
+	RPKCS7CertificateRevocationLists crls; //Optional
+	RPKCS7SignerInfos signerinfos;
+} RPKCS7SignedData;
+
+typedef struct r_pkcs7_container_t {
+	RASN1String *contentType;
+	RPKCS7SignedData signedData;
+} RPKCS7Container;
+
+R_API bool r_pkcs7_parse_certificaterevocationlists (RPKCS7CertificateRevocationLists *crls, RASN1Object *object);

```

Comment:
you can only do one operation over those. so ii dont see the reason for having over 9000 methods here.

the only operation you can do is `parse()`. and this should return a struct with all the info needed there, in linked lists, etc, in addition, a `free` method must be able to liberate that memory.